ExtraCalcFunctions.php

```diff
@@ -28,28 +29,61 @@ public function redcap_every_page_before_render( $project_id )
 		require 'functions.php';
 
 		// If auto-updating of calculated values is enabled, do this if it has not been done in the
-		// last 15 minutes.
+		// last 15 minutes (frequency is reduced if recalculations take a long time).
+		$this->needsAutoCalc = false;
+		$lastDuration = $project_id === null
+		                ? 0 : ( $this->getProjectSetting( 'calc-values-auto-update-dur' ) ?? 0 );
+		$autoCalcWait = $lastDuration < 90 ? 900 : ( $lastDuration * 10 );
```

Comment:
If I read this correctly, if it took less than 90 seconds to refresh the calculations last time it will keep the wait duration at 900 seconds (15 mins), but if more that 90 seconds then the previous wait duration is multiplied by 10.  Would this be a wait of 2.5 hours?  Isn't that rather long? And if the next times it ran, would it multiply this duration again by 10, ie 25 hours, 250 hours etc.  I may be reading it wrongly of course, but please check!

Reply:
It's not the previous wait duration that is multiplied by 10, but the previous *run* duration.
So 90 second (or less) run duration -> 900 second (15 min) wait duration.
100 second run duration -> 1000 second wait duration.
200 second run duration -> 2000 second wait duration.
etc.

Reply:
Thanks, perfect!